java/core/src/test/java/sleeper/core/iterator/AggregationFilteringIteratorTest.java

```diff
@@ -15,362 +15,362 @@
-        CloseableIterator<Row> iterator = new WrappedIterator<>(List.of(
-                new Row(Map.of("key", "test", "value", 458498L)),
-                new Row(Map.of("key", "test", "value", 87L)),
-                new Row(Map.of("key", "test", "value", 222474L))).iterator());
-
-        // When
-        List<Row> resultList = new ArrayList<>();
-        maxAggregatorIterator.apply(iterator)
-                .forEachRemaining(resultList::add);
-
-        // Then
-        assertThat(resultList).containsExactlyElementsOf(List.of(new Row(Map.of("key", "test", "value", 458498L))));
+            // Then
+            assertThat(resultList).containsExactlyElementsOf(List.of(new Row(Map.of("key", "test", "value", 458498L))));
+        }
     }
 
-    @ParameterizedTest
-    @CsvSource({"map_max", "Map_Max", "MAP_MAX"})
-    void shouldApplyMapMaxAggregationFromProperties(String aggregator) throws IteratorCreationException {
-        // Given
-        Schema schema = Schema.builder()
-                .rowKeyFields(new Field("key", new StringType()))
-                .sortKeyFields(new Field("sort", new StringType()))
-                .valueFields(new Field("value", new LongType()), new Field("map_value2", new MapType(new StringType(), new LongType())))
-                .build();
-
-        SortedRowIterator sumAggregatorIterator = new IteratorFactory(
-                new ObjectFactory(IteratorFactoryTest.class.getClassLoader()))
-                .getIterator(IteratorConfig.builder()
-                        .filteringString("")
-                        .aggregationString("sum(value)," + aggregator + "(map_value2)")
-                        .build(), schema);
-
-        CloseableIterator<Row> iterator = new WrappedIterator<>(List.of(
-                new Row(Map.of("key", "a", "sort", "b", "value", 1L, "map_value2",
-                        Map.of("map_key1", 666L, "map_key2", 11L))),
-                new Row(Map.of("key", "a", "sort", "b", "value", 2L, "map_value2",
-                        Map.of("map_key1", 245L, "map_key2", 2L))))
-                .iterator());
-
-        // When
-        List<Row> resultList = new ArrayList<>();
-        sumAggregatorIterator.apply(iterator)
-                .forEachRemaining(resultList::add);
-
-        assertThat(resultList).containsExactly(
-                new Row(Map.of("key", "a", "sort", "b", "value", 3L, "map_value2",
-                        Map.of("map_key1", 666L, "map_key2", 11L))));
+    @Nested
+    @DisplayName("Apply aggregations against map fields")
+    class ApplyMapAggregations {
+
+        @BeforeEach
+        void setUp() {
+            tableProperties.setSchema(Schema.builder()
+                    .rowKeyFields(new Field("key", new StringType()))
+                    .sortKeyFields(new Field("sort", new StringType()))
+                    .valueFields(new Field("value", new LongType()), new Field("map_value2", new MapType(new StringType(), new LongType())))
```

Comment:
Is there a reason to include a second value field in this test? It suggests it might not work if the map field was the only value field.

Reply:
I also don't think this needs a sort key, and we probably need a separate test for that.